Source/Diagnostics/ComputeDiagFunctors/BackTransformParticleFunctor.H

```diff
@@ -0,0 +1,237 @@
+    AMREX_GPU_HOST_DEVICE
+    void operator () (const DstData& dst, const SrcData& src, int i_src, int i_dst) const noexcept
+    {
+        amrex::ignore_unused(src);
+        using namespace amrex::literals;
+        // get current src position
+        amrex::ParticleReal xpnew, ypnew, zpnew;
+        m_get_position(i_src, xpnew, ypnew, zpnew);
+        const amrex::Real gamma_new_p = std::sqrt(1.0_rt + m_inv_c2*
+                                        ( m_uxpnew[i_src] * m_uxpnew[i_src]
+                                        + m_uypnew[i_src] * m_uypnew[i_src]
+                                        + m_uzpnew[i_src] * m_uzpnew[i_src]));
+        const amrex::Real gamma_old_p = std::sqrt(1.0_rt + m_inv_c2*
+                                        ( m_uxpold[i_src] * m_uxpold[i_src]
+                                        + m_uypold[i_src] * m_uypold[i_src]
+                                        + m_uzpold[i_src] * m_uzpold[i_src]));
+        const amrex::Real t_new_p = m_gammaboost * m_t_boost - m_uzfrm * zpnew * m_inv_c2;
+        const amrex::Real z_new_p = m_gammaboost* ( zpnew + m_betaboost * m_Phys_c * m_t_boost);
+        const amrex::Real uz_new_p = m_gammaboost * m_uzpnew[i_src] - gamma_new_p * m_uzfrm;
+        const amrex::Real t_old_p = m_gammaboost * (m_t_boost - m_dt)
+                                    - m_uzfrm * m_zpold[i_src] * m_inv_c2;
+        const amrex::Real z_old_p = m_gammaboost * ( m_zpold[i_src] + m_betaboost
+                                                     * m_Phys_c * (m_t_boost - m_dt ) );
+        const amrex::Real uz_old_p = m_gammaboost * m_uzpold[i_src] - gamma_old_p * m_uzfrm;
+        // interpolate in time to t_lab
+        const amrex::Real weight_old = (t_new_p - m_t_lab)
+                                     / (t_new_p - t_old_p);
+        const amrex::Real weight_new = (m_t_lab - t_old_p)
+                                     / (t_new_p - t_old_p);
+        // weighted sum of old and new values
+        const amrex::ParticleReal xp = m_xpold[i_src] * weight_old + xpnew * weight_new;
+        const amrex::ParticleReal yp = m_ypold[i_src] * weight_old + ypnew * weight_new;
+        const amrex::ParticleReal zp = z_old_p * weight_old + z_new_p * weight_new;
+        const amrex::ParticleReal uxp = m_uxpold[i_src] * weight_old
+                                      + m_uxpnew[i_src] * weight_new;
+        const amrex::ParticleReal uyp = m_uypold[i_src] * weight_old
+                                      + m_uypnew[i_src] * weight_new;
+        const amrex::ParticleReal uzp = uz_old_p * weight_old
+                                      + uz_new_p * weight_new;
+#if defined (WARPX_DIM_3D)
+        dst.m_aos[i_dst].pos(0) = xp;
+        dst.m_aos[i_dst].pos(1) = yp;
+        dst.m_aos[i_dst].pos(2) = zp;
+#elif defined (WARPX_DIM_2D)
+        dst.m_aos[i_dst].pos(0) = xp;
+        dst.m_aos[i_dst].pos(1) = zp;
+        amrex::ignore_unused(yp);
+#elif defined (WARPX_DIM_1D)
```

Comment:
1D macro is defined as WARPX_DIM_1D_Z, so this should be `#elif defined (WARPX_DIM_1D_Z)`?